vars/buildPlugin.groovy

```diff
@@ -61,7 +61,14 @@ def call(Map params = [:]) {
     }
 
     tasks[stageIdentifier] = {
+      int retryCounter = 0
       retry(count: 3, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
+        if (retryCounter > 1 && platform != 'windows') {
+          // no spot instances for windows for now TODO change when available
+          // Use a spot instance for the 2 first try and nonspot for third and last
+          label = 'ubuntu-22-amd64-maven17-nonspot'
```

Comment:
```suggestion
          label += '-nonspot'
```

to avoid fragility of the string

Reply:
but the `original` agent label is 'vm && linux' https://github.com/jenkins-infra/pipeline-library/pull/917/files#diff-db2794773b738dbb5566313f115cdf3ff1f45e13e57cbc37aa1bd181819ed5fbL55 in order to use the nonspot we define the complete label but the "spot" one is 'vm && linux' ...